src/parser/name_resolution.c

```diff
@@ -8137,7 +8138,7 @@ pt_resolve_cte_specs (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
       curr_cte->next = NULL;
 
       /* curr_cte->next should be set to NULL; next CTEs will be resolved for each previous CTE */
-      curr_cte = parser_walk_tree (parser, curr_cte, pt_resolve_spec_to_cte, cte_defs->next, NULL, NULL);
+      curr_cte = parser_walk_tree (parser, curr_cte, pt_resolve_spec_to_cte, cte_defs->next, pt_count_ctes_post, NULL);
```

Comment:
What about making a new post function to nullify `node->etc`?
It should have a part of `pt_count_ctes_post` function but it may be clearer.